### ISSUE TITLE
Component Events not working without native HTML tags

### ISSUE THREAD

**Issue by joshgarde** [marko-js/marko#807]
I'm trying to latch onto a DOM event, but I'm getting complication errors. Here's my template

```
<macro main_input_field(text, loading)>
  <div class="field">
    <div class="control">
      <button class="button${loading ? ' is-loading' : ''}" on-click("onClick")>
        ${text}
      </button>
    </div>
  </div>
</macro>

<if (input.label)>
  <div class="field is-horizontal">
    <div class="field-label is-normal">
    </div>
    <div class="field-body">
      <main_input_field text=input.text loading=input.loading/>
    </div>
  </div>
</if>
<else>
  <main_input_field text=input.text loading=input.loading/>
</else>
```

Here's my component.js:
```
module.exports = {
  onClick: function() {
    this.emit('click');
  }
}
```

Yes, I'm only using it to proxy out the button's onClick event. And here's the error:

```
web_1      | Error: Render error. Exception: ReferenceError: __component is not defined
web_1      |     at macro_main_input_field (/app/app/components/input_button/index.marko.js:18:18)
web_1      |     at render (/app/app/components/input_button/index.marko.js:32:5)
web_1      |     at renderCompontent (/app/node_modules/marko/src/components/taglib/helpers/renderComponent.js:3:5)
```

If I attach an empty `<span></span>` to the end of the template, it compiles just fine, but without it, complication fails.

**Comment by patrick-steele-idem** [comment]
Typing on phone so sorry for lack of details, but we are testing a fix that allows UI components to have multiple root elements, dynamic root elements, root text elements, etc. This will resolve this issue and other related issues. We should have a fix published by the end of the week and will close this issue when released. For the short-term, fixed root HTML elements are required.

**Comment by austinkelleher** [comment]
This should be resolved in the latest beta `marko@v4.5.0-beta.4`. We should release these changes out of beta sometime next week. Please feel free to reopen this issue if you continue to see issues.
